mne/source_space.py

```diff
@@ -1527,24 +1534,38 @@ def setup_volume_source_space(subject, fname=None, pos=5.0, mri=None,
             surf['rr'] += sphere[:3] / 1000.0  # move by center
             complete_surface_info(surf, do_neighbor_vert=True, copy=False)
         # Make the grid of sources in MRI space
-        sp = _make_volume_source_space(surf, pos, exclude, mindist, mri,
-                                       volume_label)
+        if volume_label is not None:
+            sp = []
+            for label in volume_label:
+                vol_sp = _make_volume_source_space(surf, pos, exclude, mindist,
+                                                   mri, label)
+                sp.append(vol_sp)
+        else:
+            sp = _make_volume_source_space(surf, pos, exclude, mindist, mri,
+                                           volume_label)
 
     # Compute an interpolation matrix to show data in MRI_VOXEL coord frame
+    if not isinstance(sp, list):
+        sp = [sp]
+
     if mri is not None:
-        _add_interpolator(sp, mri, add_interpolator)
-    elif sp['type'] == 'vol':
+        for s in sp:
+            _add_interpolator(s, mri, add_interpolator)
+    elif sp[0]['type'] == 'vol':
         # If there is no interpolator, it's actually a discrete source space
-        sp['type'] = 'discrete'
+        sp[0]['type'] = 'discrete'
```

Comment:
Reading the code I understood there was only a volume src space  
`sp['type'] = 'discrete'
for this I set `sp[0]['type'] = 'discrete'` 
since now sp is a list